src/Umbraco.Core/Security/Authorization/IAuthorizationHelper.cs

```diff
@@ -9,11 +10,19 @@ namespace Umbraco.Cms.Core.Security.Authorization;
 public interface IAuthorizationHelper
 {
     /// <summary>
-    ///     Converts an <see cref="IUser" /> into <see cref="IPrincipal" />.
+    ///     Converts an <see cref="IPrincipal" /> into an <see cref="IUser" />.
     /// </summary>
     /// <param name="currentUser">The current user's principal.</param>
     /// <returns>
     ///     <see cref="IUser" />.
     /// </returns>
     IUser GetUmbracoUser(IPrincipal currentUser);
+
+    /// <summary>
+    ///     Attempts to convert an <see cref="IPrincipal" /> into an <see cref="IUser" />.
+    /// </summary>
+    /// <param name="currentUser">The current user's principal.</param>
+    /// <param name="user">The resulting <see cref="IUser" />, if the conversion is successful.</param>
+    /// <returns>True if the conversion is successful, false otherwise</returns>
+    bool TryGetUmbracoUser(IPrincipal currentUser, [NotNullWhen(true)] out IUser? user);
```

Comment:
How is this not breaking. ? 

Do we mis something to check for breaking changes?

I think we should make an interface implementation just catching the exception from GetUmbracoUser

@kjac @Zeegaan

Reply:
You are right, seems some package validation is missing if it's not catching these things automatically 😢

Reply:
Fix here: https://github.com/umbraco/Umbraco-CMS/pull/16899